lnwire/features.go

```diff
@@ -0,0 +1,175 @@
+package lnwire
+
+import (
+	"github.com/go-errors/errors"
+	"math"
+	"sync/atomic"
+)
+
+// featureFlag represent the status of the feature optional/required and needed
+// to allows future incompatible changes, or backward compatible changes.
+type featureFlag uint8
+
+// featureName represent the name of the feature and needed in order to have the
+// compile errors if we specify wrong feature name.
+type featureName string
+
+type NameToIndex map[featureName]int
+type indexToFlag map[int]featureFlag
+
+const (
+	// optionalFlag represent the feature which we might be used already but
+	// it isn't required yet, and if remote peer doesn't have this feature
+	// we may turn it off without disconnecting with peer.
+	optionalFlag featureFlag = 2 // 0b10
+
+	// requiredFlag represent the features which is required for proper
+	// peer interaction, we disconnect with peer if it doesn't have this
+	// particular feature.
+	requiredFlag featureFlag = 3 // 0b11
```

Comment:
I know we've gone back and forth on this several times at this point, but shouldn't this be: `0b01`?